### ISSUE TITLE
Markup/Comment Solution

### ISSUE THREAD

**Issue by apat183** [box/viewer.js#225]
Hi,

I'm new to viewer. I'm looking for a markup/comment solution for viewer.js. I understand the only crocodoc version of veiwer had this capability.

https://crocodoc.com/docs/walkthrough/comments/

Has anyone implemented a similar solution to comments as above? Or come up with an alternative solution? 

I was also looking at https://github.com/openannotation/annotator, anyone used this with viewer? I would rather something within viewer instead though.

Thanks

**Comment by tonyjin** [comment]
@apat183 could you describe your use case a bit more? If you're interested in annotations on documents, we (at Box) are working on a solution, assuming you're okay with having your documents stored on Box.

**Comment by apat183** [comment]
@tonyjin Current use case is person X uploads document (docx/xlsx) for review on webapp to person Y. I use BOX viewer api to convert docx/xlsx for preview in site. I currently use the box viewer api to also store the converted contents on AWS S3 (because I need more control on structure of how previews are stored). Person Y then reviewers preview of document in webapp, they need to be able to provide feedback on document, either changes that need to be made etc. The old way linked here https://crocodoc.com/docs/walkthrough/comments/ is perfect for what I need to do, add comments to specific sections, cross text out and highlight sections. Person x would then see markups/comments and make changes to offline version and upload and replace.

Could be interested in storing documents in BOX, since the webapp has sensitive documents and box is HIPPA compliant etc. Only concern is I don't want to divert users to BOX etc, are you talking about an integrated solution like viewer? and can it be used for word and excel documents?

**Comment by priyajeet** [comment]
Yes, the replacement for viewer.js will be something that can talk to Box via the APIs and will be supporting all files types https://box-content.readme.io/reference#get-embed-link

**Comment by apat183** [comment]
@priyajeet Awesome thanks, any timeframe on comments part? I looked at the content api and I can already start embedding the folders and files preview.
